# HG changeset patch
# User Philip Nienhuis <prnienhuis@users.sf.net>
# Date 1429284865 25200
#      Fri Apr 17 08:34:25 2015 -0700
# Node ID d6e523825abc61bb6981ba95d20738f52df7ed60
# Parent  8bfadb26afcd86459adae6f4d3065bd6fcc22abd
zip.m: fix XFAIL on Windows; fix test (bug #44811).

* zip.m: Don't use multiple commands in one system call since shell cmd
separator is different between Windows and UNIX.  Correct "extension" to
"ext" to match real variable name in BIST test.

diff --git a/scripts/miscellaneous/zip.m b/scripts/miscellaneous/zip.m
--- a/scripts/miscellaneous/zip.m
+++ b/scripts/miscellaneous/zip.m
@@ -50,20 +50,22 @@ function filelist = zip (zipfile, files,
   elseif (! iscellstr (files))
     error ("zip: FILES must be a character array or cellstr");
   endif
 
   rootdir = tilde_expand (rootdir);
 
   zipfile = make_absolute_filename (zipfile);
 
-  cmd = sprintf ("cd %s; zip -r %s %s",
-                     rootdir,   zipfile, sprintf (" %s", files{:}));
+  cmd = sprintf ("zip -r %s %s", zipfile, sprintf (" %s", files{:}));
 
+  origdir = pwd ();
+  cd (rootdir);
   [status, output] = system (cmd);
+  cd (origdir);
 
   if (status)
     error ("zip: zip failed with exit status = %d", status);
   endif
 
   if (nargout > 0)
     cmd = ["unzip -Z -1 " zipfile];
     [status, filelist] = system (cmd);
@@ -103,17 +105,17 @@ endfunction
 %!   assert (fid >= 0);
 %!   new_data = fread (fid);
 %!   fclose (fid);
 %!   if (orig_data != new_data)
 %!     error ("unzipped file not equal to original file!");
 %!   endif
 %! unwind_protect_cleanup
 %!   unlink (filename);
-%!   unlink ([dirname, filesep, basename, extension]);
+%!   unlink ([dirname, filesep, basename, ext]);
 %!   rmdir (dirname);
 %! end_unwind_protect
 
 ## Test input validation
 %!error zip ()
 %!error zip (1)
 %!error zip (1,2,3,4)
 %!error <ZIPFILE must be a string> zip (1, "foobar")
